pkg/armadaevents/events.proto

```diff
@@ -290,6 +290,8 @@ message JobRunLeased {
     string node_id = 4;
     // Used by the scheduler to maintain a consistent state
     int32 update_sequence_number = 5;
+    // Additional labels to be added to the PodSpec.
+    map<string, string> additional_labels = 6;
```

Comment:
Do we really want to use labels instead of annotations?

Reply:
I don't have a preference really - if an annotation makes more sense happy to use that

Reply:
The only reason we went for a label is is because we currently use those for similar things like job id / jobset / queue id.

Reply:
Looking at https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/#attaching-metadata-to-objects, using an annotation might be more appropriate. Do we currently set any gang scheduling metadata on gang jobs (e.g., gang ID would arguably fall into the label category)?

Reply:
Cool - I'll rename this to annotations then as it's not really something we want to select/identify with.

DogeFarm sets GangId etc. as metadata in DogeFarm and as an annotation to Armada

E.g. Yaml spec:
annotations:
      armadaproject.io/gangCardinality: '4'
      armadaproject.io/gangId: 28cf33a61192490a87974b0ad6eeb698-7